src/lib/gckms.ts

```diff
@@ -0,0 +1,48 @@
+import kms from "@google-cloud/kms";
+import { Storage } from "@google-cloud/storage";
+
+export interface KeyConfig {
+    projectId: string;
+    locationId: string;
+    keyRingId: string;
+    cryptoKeyId: string;
+    ciphertextBucket: string;
+    ciphertextFilename: string;
+}
+
+const { GCP_STORAGE_CONFIG } = process.env;
+
+// Allows the environment to customize the config that's used to interact with google cloud storage.
+// Relevant options can be found here: https://googleapis.dev/nodejs/storage/latest/global.html#StorageOptions.
+// Specific fields of interest:
+// - timeout: allows the env to set the timeout for all http requests.
+// - retryOptions: object that allows the caller to specify how the library retries.
+const storageConfig = GCP_STORAGE_CONFIG ? JSON.parse(GCP_STORAGE_CONFIG) : undefined;
+
+// This function takes an array of GCKMS configs that are shaped as follows:
+// {
+//   projectId: "project-name",
+//   locationId: "asia-east2",
+//   keyRingId: "Keyring_Test",
+//   cryptoKeyId: "keyname",
+//   ciphertextBucket: "cipher_bucket",
+//   ciphertextFilename: "ciphertext_fname.enc",
+// }
+//
+// It returns a private key that can be used to send transactions.
+export async function retrieveGckmsKey(gckmsConfig: KeyConfig): Promise<string> {
```

Comment:
I had to add this code here again, as @uma/common has ethers v5, incompatible with this project.